ank/src/cli_commands/workload_table.rs

```diff
@@ -0,0 +1,304 @@
+#[cfg(not(test))]
+use crate::log::terminal_width;
+
+#[cfg(test)]
+fn terminal_width() -> usize {
+    80
+}
+
+use super::workload_table_row::{ColumnPosition, MaxAdditionalInfo};
+use tabled::{
+    settings::{object::Columns, Modify, Padding, Style, Width},
+    Table, Tabled,
+};
+pub struct WorkloadTable<RowType> {
+    rows: Vec<RowType>,
+}
+
+impl<RowType> WorkloadTable<RowType>
+where
+    RowType: Tabled + ColumnPosition,
+    Vec<RowType>: MaxAdditionalInfo,
+{
+    const ADDITIONAL_INFO_SUFFIX: &'static str = "...";
+
+    pub fn new(rows: Vec<RowType>) -> Self {
+        Self { rows }
+    }
+
+    // [impl->swdd~cli-shall-present-workloads-as-table~1]
+    pub fn create_default_table(&self) -> String {
+        let default_table = Self::default_table(&self.rows);
+        default_table.to_string()
+    }
+
+    // [impl->swdd~cli-shall-present-workloads-as-table~1]
+    pub fn create_table_truncated_additional_info(&self) -> Option<String> {
+        let default_table = Self::default_table(&self.rows);
+
+        let total_table_width: usize = default_table.total_width();
+        let remaining_terminal_width =
```

Comment:
We should maybe rename for better readability: `remaining_terminal_width` -> `available_additional_info_width`

Reply:
Done